Source/WTF/wtf/glib/GSpanExtras.h

```diff
@@ -19,11 +19,40 @@
 
 #pragma once
 
+#include <wtf/MallocSpan.h>
 #include <wtf/StdLibExtras.h>
 #include <wtf/glib/GRefPtr.h>
+#include <wtf/glib/GUniquePtr.h>
+
+extern "C" {
+typedef char** GStrv;
+void g_strfreev(GStrv);
+}
 
 namespace WTF {
 
+template <typename T>
+struct GMalloc {
+    static void free(T* ptr) { g_free(ptr); }
+};
+
+template <>
+struct GMalloc<GStrv> {
+    static void free(GStrv ptr) { g_strfreev(ptr); }
```

Comment:
```suggestion
struct GMalloc<char**> {
    static void free(char** ptr) { g_strfreev(ptr); }
```

Reply:
Here I would like to actually have a way of telling apart whether `g_free()` or `g_strfreev()` should be used. By specializing on `GMalloc<char**>` (or `GMalloc<GStrv>`) we would always pick the specialization that uses `g_strfreev()` when instantiating `GMallocSpan<char*>` without having the chance of picking the `g_free()` one&mdash;which may be needed if we want to wrap something returned by GLib which is a `char**` with `(transfer container)`!

So what I am going to do is plainly:

```cpp
struct GMalloc {
    static void free(void* ptr) { g_free(ptr); }
};

struct GMallocStrv {
    static void free(void* ptr) { g_strfreev(ptr); }
};

template <typename T, typename Malloc = GMalloc>
using GMallocSpan = MallocSpan<T, Malloc>;
```

This way by default a `GMallocSpan<char*>` will use `GMalloc`, and we can specifically opt-in to use `g_strfreev()` using a `GMallocSpan<char*, GMallocStrv>`.
```